src/SFA.DAS.LevyTransferMatching.Api/Controllers/FunctionsController.cs

```diff
@@ -104,5 +105,32 @@ public async Task<IActionResult> DebitApplication(DebitApplicationRequest reques
                 return new StatusCodeResult((int)HttpStatusCode.InternalServerError);
             }
         }
+
+        [Route("application-funding-declined")]
+        [HttpPost]
+        public async Task<IActionResult> ApplicationFundingDeclined(ApplicationFundingDeclinedRequest request)
+        {
+            try
+            {
+                var result = await _mediator.Send(new CreditPledgeCommand
```

Comment:
I think we need a small amount of logic adding, and this looks like the best place to put it.

If the request.Amount is zero, then we shouldn't call the Credit endpoint - not only would it achieve nothing, there's a validation rule that prevents zero-valued credits. (Amount can be zero since it's the cost in the current f/y, and an April start will cost nothing).

We might have missed this on the Debit use-case, so I'll add a tech debt task to deal with it.

Reply:
I've added logic in to check the amount - if it's zero or lower, then the inner API isn't called.